### PR TITLE
Long term storage

### PR DESCRIPTION
**Pull request by 1fbff5f83b23d39d38b1dfcb4cac8d9b**
Didn't roll antag? Feeling like a dick for wasting a job slot while looking for murderbone ghost roles? Fear no more!
Introducing the cryptosleep casket! Simply remove any important and unique (traitor objective) items and place them in a secure spot, then stand in front of the machine and activate it! Accidental death and dismemberment insurance sold separately.
The casket will delete your character and free up a job slot for other players to sign up as.

Image of the room formerly known as Cabin 1:
image

:cl: McDonald072
rscadd: Considering suicide or going braindead? Go to dorms and enter long term storage instead. Your job slot will become available to latejoin players.
/:cl:

Current item blacklist:
- Captain's antique laser
- HOS' laser gun
- Hand tele
- Captain's jetpack
- Medal of captainacy
- CMO's hypospray
- Nuclear auth disk
- Reflector armor and helmet
- RD's teleport armor
- Secret documents of any organization
- Nuke core (as if)
- Antique plasma tank
- Station blueprints
- Rapid Engineering device
- Pinpointer
- All spacesuits
- CE's advanced magboots
- Security magboots
- Captain's spare ID
- Janitor's key

### CODE REVIEW

File: code/game/objects/structures/longtermstorage.dm [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		say("Please remove those items before proceeding.")
+		return
+	if(alert(user, "Are you sure you want to enter storage? This action cannot be undone!", , "Yes", "No") == "No") return
+	target.loc = src
+	icon_state = "casket-storing"
+	sleep(10)
+	playsound(src.loc, 'sound/machines/juicer.ogg', 100, 1)
+
+	var/datum/job/j = SSjob.GetJob(target.mind.assigned_role)
+	if(j) j.current_positions--
```
Comment (Jc1469) [inline code/game/objects/structures/longtermstorage.dm]: what is this unnatural coding practice

File: code/game/objects/structures/longtermstorage.dm [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+	var/msg = "[target]"
+	if(j) msg += ", [target.mind.assigned_role], "
```
Comment (Jc1469) [inline code/game/objects/structures/longtermstorage.dm]: every fibre of my being having done byond code hates this.  the other parts don't mind, the byond part no like

File: code/game/objects/structures/longtermstorage.dm [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	for(var/I in forbidden_items)
+		var/obj/item/FI = locate(I) in all_items
+		if(FI) found += FI
+	if(found.len)
+		say("The following items have been found on your person:")
+		for(var/obj/item/I in found)
+			say("[I].")
+		say("Please remove those items before proceeding.")
+		return
+	if(alert(user, "Are you sure you want to enter storage? This action cannot be undone!", , "Yes", "No") == "No") return
```
Comment (Jc1469) [inline code/game/objects/structures/longtermstorage.dm]: newline for return :badminton:

File: code/game/objects/structures/longtermstorage.dm [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	var/mob/living/carbon/human/target = locate() in get_step(src, EAST)
+	if(!target)
+		say("Please step in front of the machine before starting the cryptosleep process.")
+		return
+	if(target != user) return //no forceful storage
+	var/list/found = list()
+	var/list/all_items = target.GetAllContents()
+	for(var/I in forbidden_items)
+		var/obj/item/FI = locate(I) in all_items
+		if(FI) found += FI
```
Comment (Jc1469) [inline code/game/objects/structures/longtermstorage.dm]: wake me up inside

**Review by Jc1469 — CHANGES_REQUESTED** [review]
To add, that thing I said about using high_risk instead of a list of blacklisted things.

File: code/game/objects/structures/longtermstorage.dm (hunk 4 above)
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [reply code/game/objects/structures/longtermstorage.dm]: remnant of the way i checked items before

File: code/game/objects/structures/longtermstorage.dm (hunk 3 above)
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [reply code/game/objects/structures/longtermstorage.dm]: no

File: code/game/objects/structures/longtermstorage.dm (hunk 1 above)
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [reply code/game/objects/structures/longtermstorage.dm]: my dad beats me every 50 lines of code

File: code/game/objects/structures/longtermstorage.dm (hunk 2 above)
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [reply code/game/objects/structures/longtermstorage.dm]: sigh fine

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
I don't  want to use high risk items cause some of these are not high risk yet they shouldn't be deleted, like spacesuits

**Comment by Jc1469** [comment]
fair enough.

**Comment by Xantam-13** [comment]
Good!

**Comment by Jc1469** [comment]
I await the inevitable grief where people dismantle R&D or something then go and delete themselves with all the boards

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
it's just as easy to space R&D or something

**Comment by Ratvar** [comment]
Can you delete posibrains, soulshards, MMI's and such shit this way? You cannot just kill posi and it still can radio from shitty places, and those interactions are commonly bugged

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
it should delete everything you have in your inventory when you enter it, i'll need to test that

**Comment by KrabSpider** [comment]
Will there also be a console that dispenses

**Comment by Kmc2000** [comment]
Perhaps it could disposal the items to cargo instead? thinking about assustants grabbing all the cuffs and guns from the armoury and then deleting themselves. Thisll also need rules to acompany it

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
I don't think that is necessary. If an assistant can grab guns and delete themselves they might as well space them or throw them into the grinder and continue griefing.

**Comment by KrabSpider** [comment]
Has it been tested yet?

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
@AshCorr No Merge Conflict :^)

**Comment by AshCorr** [comment]
You've fucked up the map now @1fbff5f83b23d39d38b1dfcb4cac8d9b , make sure you use the mapmerge tool.

**Pull request closed without merging**